new/analyse/create-dependency-tree.ts

```diff
@@ -0,0 +1,62 @@
+import { Module } from '../types';
+import { clone } from 'ramda';
+import { yellow } from 'chalk';
+
+import { logger } from '../helpers/logger';
+
+const { error, log } = logger('[analyse] create dependency tree');
+
+export const createDependencyTree: Module = (env) => {
+  if (!env.packages) {
+    const errorMessage = 'no packages found on env config object';
```

Comment:
in general, that's a lot of boilerplate code in each module. I'd suggest something like `assertEnv(!!env.packages, "no packages found on env config object")` - assertEnv being a helper function that takes care of throwing.

Reply:
Won't be done in this PR.